File: airbyte-notification/src/main/java/io/airbyte/notification/FakeCustomerIoEmailNotificationSender.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.notification;

import jakarta.inject.Singleton;
import java.lang.invoke.MethodHandles;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Fake client for customerIoNotification - this will not send any actual emails.
 */
@Singleton
public class FakeCustomerIoEmailNotificationSender extends CustomerIoEmailNotificationSender {

  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());

  FakeCustomerIoEmailNotificationSender() {
    super(null, null);
  }

  @Override
  protected void callCustomerIoSendNotification(final String custumerIoPayload) {
    log.info("FakeCustomerIoEmailNotificationSender: callCustomerIoSendNotification: {}", custumerIoPayload);
  }

}

File: airbyte-notification/src/main/kotlin/io/airbyte/notification/FakeCustomerIoEmailNotificationSender.kt
/*
 * Copyright (c) 2020-2026 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.notification

import io.airbyte.metrics.MetricClient
import io.airbyte.micronaut.runtime.AirbyteNotificationConfig
import io.github.oshai.kotlinlogging.KotlinLogging
import jakarta.inject.Singleton
import okhttp3.OkHttpClient
import java.util.UUID

private val log = KotlinLogging.logger { }

/**
 * Fake client for customerIoNotification - this will not send any actual emails.
 */
@Singleton
class FakeCustomerIoEmailNotificationSender internal constructor() :
  CustomerIoEmailNotificationSender(OkHttpClient(), AirbyteNotificationConfig(), MetricClient(null)) {
    override fun callCustomerIoSendNotification(
      customerIoPayload: String,
      workspaceId: UUID?,
    ) {
      log.info { "FakeCustomerIoEmailNotificationSender: callCustomerIoSendNotification: $customerIoPayload" }
    }
  }
